core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java

```diff
@@ -16,45 +16,44 @@ public void whenFloatComparedWithDifferentValues_thenCouldMatch() {
     }
 
     @Test
-    public void whenCreatedFromCertainFloatValues_thenMatches() {
-        float floatToConvert = 0.5f;
+    public void whenCreatedFromFloat_thenMatchesInternallyStoredValue() {
+        float floatToConvert = 1.1f;
         BigDecimal bdFromFloat = new BigDecimal(floatToConvert);
-        assertEquals("0.5", bdFromFloat.toString());
+        assertEquals("1.10000002384185791015625", bdFromFloat.toString());
     }
 
     @Test
-    public void whenCreatedFromCertainFloatValues_thenDoesNotMatch() {
-        float floatToConvert = 1.1f;
-        BigDecimal bdFromFloat = new BigDecimal(floatToConvert);
-        assertEquals("1.10000002384185791015625", bdFromFloat.toString());
+    public void whenCreatedFromString_thenPreservesTheOriginal() {
+        BigDecimal bdFromString = new BigDecimal("1.1");
+        assertEquals("1.1", bdFromString.toString());
     }
 
     @Test
-    public void whenCreatedFromString_thenMatches() {
+    public void whenCreatedFromFloatConvertedToString_thenFloatInternalValueGetsTruncated() {
         String floatValue = Float.toString(1.1f);
         BigDecimal bdFromString = new BigDecimal(floatValue);
+        assertEquals("1.1", floatValue);
         assertEquals("1.1", bdFromString.toString());
     }
 
     @Test
-    public void whenCreatedByValueOfAndIsFloat_thenDoesNotMatch() {
-        float floatToConvert = 1.1f;
-        BigDecimal bdByValueOf = BigDecimal.valueOf(floatToConvert);
-        assertEquals("1.100000023841858", bdByValueOf.toString());
+    public void whenFloatConvertedToString_thenGetsTruncated() {
+        String floatValue = Float.toString(1.10000002384185791015625f);
+        assertEquals("1.1", floatValue);
     }
 
     @Test
-    public void whenFloatCastToDouble_thenGotADifferentNumber() {
-        float floatToConvert = 1.1f;
-        double doubleCast = floatToConvert;
-        assertEquals("1.100000023841858", Double.toString(doubleCast));
+    public void whenCreatedByValueOf_thenFloatValueGetsTruncated() {
+        assertEquals("1.100000023841858", BigDecimal.valueOf(1.1f)
+            .toString());
+        assertEquals("1.100000023841858", BigDecimal.valueOf(1.10000002384185791015625f)
+            .toString());
```

Comment:
Can we format each of these on a single line, or do you prefer it the way it is?

Reply:
Ok, I fixed the format.

Reply:
I changed the test cases a little to align with the article updates

Reply:
I changed the string constructor test case with the clearer "1.1" input value